workers/unity/Assets/Gdk/Core/Collections/ListProvider.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+
+namespace Assets.Gdk.Core.Collections
+{
+    internal static class ListProvider<T>
+    {
+        private static readonly Dictionary<uint, List<T>> Storage = new Dictionary<uint, List<T>>();
+
+        internal static void AllocateHandle(uint handle)
+        {
+            var item = new List<T>();
+            Storage.Add(handle, item);
+        }
+
+        internal static List<T> Get(uint handle)
+        {
+            return Storage[handle];
```

Comment:
If error checking on the free, then we should probably check here. I'd say they are about the same risk.